### PR TITLE
Compute Console Command

### PR DESCRIPTION
**Pull request by rudihorn**
This is to be merged in after #1489. It adds a console command called `compute`, which allows individual registered memoized functions to be run for debugging purposes.

For example, to call the load directory function on the "src" directory run `dune compute load-dir src`.

### CODE REVIEW

**Comment by diml** [comment]
This is ready for review

**Review by rgrinberg — APPROVED** [review]

File: src/build_system.ml [hunk 1]
```diff
@@ -89,6 +89,11 @@ module Internal_rule = struct
 
   let loc ~file_tree ~dir t  = rule_loc ~file_tree ~dir ~loc:t.loc
 
+  let decode : t Dune_lang.Decoder.t =
+    let open Dune_lang.Decoder in
+    loc >>= fun loc ->
+    Errors.fail loc "<not-implemented>"
```
Comment (emillon) [inline src/build_system.ml]: If this is failing decoder is specific to memoization I think it's be better to move it to the `Memo.Make` argument.

Comment (diml) [reply src/build_system.ml]: Indeed, I reworked a bit the API

**Comment by diml** [comment]
I also removed `dune list-functions` and replaced it by `dune compute list`

File: src/build_system.ml (hunk 1 above)
Comment (emillon) [reply src/build_system.ml]: thanks!

**Review by emillon — APPROVED** [review]

**Comment by diml** [comment]
I pressed "rebase" instead of "squash" so the _ commits went into master /o\

**Pull request merged**